core/version/gear.ex

```diff
@@ -113,63 +115,92 @@ defmodule AntikytheraCore.Version.Gear do
   @typep gear_dependencies :: MapSet.t()
   @typep gear_and_deps_pair :: {GearName.t(), gear_dependencies}
 
-  defun install_gears_at_startup(gear_names :: v[[GearName.t()]]) :: :ok do
+  defun install_gears_at_startup(gear_names :: v[[GearName.t()]]) :: :ok | :error do
     if Antikythera.Env.running_with_release?() do
       do_install_gears_at_startup(gear_names)
     else
       :ok
     end
   end
 
-  defunp do_install_gears_at_startup(gear_names :: v[[GearName.t()]]) :: :ok do
+  defunpt do_install_gears_at_startup(gear_names :: v[[GearName.t()]]) :: :ok | :error do
+    # Tests need `__MODULE__.` to mock these functions.
     gear_and_deps_pairs =
-      Enum.map(gear_names, fn g -> {g, gear_dependencies_from_app_file(g, gear_names)} end)
-
-    pairs_not_installed =
-      install_gears_whose_deps_met(gear_and_deps_pairs, MapSet.new(), fn gear_name ->
-        try do
-          install_or_upgrade_to_next_version(gear_name)
-        rescue
-          e -> L.error("Failed to install #{gear_name}: #{Exception.message(e)}")
-        end
+      Enum.map(gear_names, fn g ->
+        {g, __MODULE__.gear_dependencies_from_app_file(g, gear_names)}
       end)
 
+    {pairs_not_installed, num_failed_install} =
+      __MODULE__.install_gears_whose_deps_met(
+        gear_and_deps_pairs,
+        MapSet.new(),
+        0,
+        fn gear_name ->
+          try do
+            install_or_upgrade_to_next_version(gear_name)
+          rescue
+            e ->
+              L.error("Failed to install #{gear_name}: #{Exception.message(e)}")
+              :error
+          end
+        end
+      )
+
     Enum.each(pairs_not_installed, fn {gear_name, deps} ->
       L.error("#{gear_name} is not installed due to unmatched dependencies: #{inspect(deps)}")
     end)
+
+    num_installable = length(gear_names)
+    num_installed = num_installable - length(pairs_not_installed) - num_failed_install
+    # version_upgrade_test starts Antikythera without gears.
+    # Then, we can't check number of installed gears.
+    needs_check = Antikythera.Env.runtime_env() != :local
+
+    is_enough_gears_installed =
+      num_installable != 0 && num_installed / num_installable > @installed_gear_ratio_threshold
```

Comment:
I think `and` operator is better if both sides of the operator are expected to be boolean.
```suggestion
      num_installable != 0 and num_installed / num_installable > @installed_gear_ratio_threshold
```

Reply:
Fixed at https://github.com/access-company/antikythera/pull/152/commits/2473a91ba577ab05ec7b6ba9607577074ebe4d01